sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/Crashes.java

```diff
@@ -577,6 +578,11 @@ void saveUncaughtException(Thread thread, Throwable exception) {
             saveErrorLog(errorLog, errorStorageDirectory, filename);
             File throwableFile = new File(errorStorageDirectory, filename + ErrorLogHelper.THROWABLE_FILE_EXTENSION);
             StorageHelper.InternalStorage.writeObject(throwableFile, exception);
+
+            if (WrapperSdkExceptionManager.hasManagedExceptionData()) {
+                WrapperSdkExceptionManager.saveManagedExceptionData(errorLog.getId());
+            }
+
```

Comment:
Hum, Xamarin should not attempt to save anything before `onCrashCaptured` is called: if the callback is not called, we could leak a file. This method should not be called by Android itself but by Xamarin inside the callback (same place when we overwrite the existing json file).